### PR TITLE
Add ES 7 support

### PR DESCRIPTION
**Pull request by przbadu**
#708 

support for Elastic Search 7.x mappings

### CODE REVIEW

**Comment by silva96** [comment]
will this break the compatibility with the older versions? it might need a condition to verify how does the response format look like doesn't it?

**Comment by subelsky** [comment]
I'm testing Chewy locally and am successfully using `przbadu`'s code w/ an elasticsearch 7.0 FWIW. 👍  this was a lifesaver!

File: lib/chewy/fields/root.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def mappings_hash
         end
 
         mappings[name][:_parent] = parent.is_a?(Hash) ? parent : {type: parent} if parent
-        mappings
+        mappings[name]
```
Comment (darkleaf) [inline lib/chewy/fields/root.rb:33]: It is the breaking change. 
If you want to support both 6 and 7 Elastic versions you could pass special request params.

 https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#hits-total-now-object-search-response
 https://www.elastic.co/blog/moving-from-types-to-typeless-apis-in-elasticsearch-7-0
 https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#_schedule_for_removal_of_mapping_types

https://github.com/toptal/chewy/issues/673#issuecomment-631432562

**Review by przbadu — COMMENTED** [review]
But I think, because it is a breaking change in ElasticSearch version, maybe we can have newer version of chewy handling ES 7+ and we can have previous version for old ES version?

**Comment by TrevorHinesley** [comment]
Unfortunately, this doesn't allow querying on types in an index. 

https://github.com/toptal/chewy#composing-requests

^ the example of `PlaceIndex::City` below will return no results:

```ruby
PlaceIndex.query(match: {name: 'London'}) # returns documents of any type
PlaceIndex::City.query(match: {name: 'London'}) # returns cities only.
```

**Comment by bbatsov** [comment]
I'm willing to accept this patch, but only if it doesn't break compatibility with older ES versions.

File: lib/chewy/fields/root.rb (hunk 1 above)
Comment (silva96) [reply lib/chewy/fields/root.rb:33]: How does the response look like for pre-7 and for 7+ ? based on that, you could determine if you need to return either `mapping` or `mapping[name]` 

or am I oversimplifying the problem and I'm not taking something in account?

**Comment by rabotyaga** [comment]
Please take a look at [Chewy 7.0.0 release](https://github.com/toptal/chewy/releases/tag/v7.0.0). Stay tuned for the 7.1 release, with dropped support for ES6.8 and removed deprecation warnings.

**Pull request closed without merging**